docs/_docs/reference/experimental/anonymous-context-bounds.md

```diff
@@ -0,0 +1,75 @@
+---
+layout: doc-page
+title: "Anonymous Context Bounds"
+---
+
+This page describes experimental support for anonymous context bounds Scala 3, available in 3.2 or later. It is enabled by the language import
+```scala
+import language.experimental.anonymousContextBounds
+```
+The reason for publishing this extension now is to get feedback on its usefulness.
+
+## Why Anonymous Context Bounds?
+
+Typeclasses are [first-class consideration of Scala 3](https://docs.scala-lang.org/scala3/book/ca-type-classes.html) 
+and [context bound sugar](https://docs.scala-lang.org/scala3/book/ca-context-bounds.html) already exists to support their use.
+Taking an example from [here](https://docs.scala-lang.org/scala3/book/ca-type-classes.html), context bounds allow one
+to write
+```scala
+def showAll[S: Showable](xs: Set[S]): Unit = xs.foreach(x => println(x.show))
+```
+
+instead of
+
+```scala
+def showAll[S](xs: Set[S])(using Showable[S]): Unit = xs.foreach(x => println(x.show))
+```
+
+See [this StackOverflow answer](https://stackoverflow.com/questions/4465948/what-are-scala-context-and-view-bounds/4467012#4467012)
+for more discussion and motivation. At the same time, Scala 3 has sought to avoid forcing the user to 
+[provide names that are never referenced](https://docs.scala-lang.org/scala3/reference/contextual/). When using
+typeclasses, it is often the case that the type parameter is never referenced -- all that matters is that
+that the compiler can prove that a `given` instance is available for that type. The `anonymousContextBounds` extension
+allows the user to avoid specifying a name for the type of a parameter with a context bound:
+
+```scala
+def showAll(xs: Set[? : Showable]): Unit = xs.foreach(x => println(x.show))
+```
+
+is sugar for
+
+```scala
+def showAll[S: Showable](xs: Set[S]): Unit = xs.foreach(x => println(x.show))
+```
+
+which further desugars as above. Rust, which also has first-class support for typeclasses,
+has a similar syntax with its [`impl` traits](https://doc.rust-lang.org/reference/types/impl-trait.html)
+
+
+## Syntax
+
+The syntax closely follows existing syntax for type bounds: note the similarity to 
+```scala
+def showAll(xs: Set[? <: Showable2]): Unit = xs.foreach(x => println(x.show))
+```
+
+and
+
+```scala
+def showAll[S <: Showable2](xs: Set[S]): Unit = xs.foreach(x => println(x.show))
+```
+
+in the case where `Showable2` is a simple trait instead of one designed to be used as a typeclass.
+
+Unlike anonymous type bounds, anonymous context bounds can appear as the top-level type of a parameter
+
+```scala
+def showOne(x: ? : Showable): Unit = x.show // okay
+def showOne[S <: Showable2](x: S): Unit = x.show // okay
+def showOne(x: ? <: Showable2): Unit = x.show // error: Unbound wildcard type
+```
+
+This is because `def showOne(x: ? <: Showable2)` is not meaningfully different from `def showOne(x: Showable2)`,
```

Comment:
Im not sure this analogy works, because if `def showOne(x: ? <: Showable2)` meant the same as `def showOne[S <: Showable2](x: S): S` then potentially we get a more refined result type than just Showable2

Reply:
Okay, but why is it disallowed then? Should I just leave that as a mystery?

Reply:
I just removed the sentence completely.